ament_index_python/test/test_ament_index_python.py

```diff
@@ -181,4 +196,110 @@ def get_package_share_directory_test(package_name, expect_prefix):
     except PackageNotFoundError:
         pass
     except Exception as exc:
-        assert False, 'Expected PackageNotFoundError, got: {}'.format(type(exc))
+        assert False, 'Expected PackageNotFoundError, got: {}' \
+            .format(type(exc))
+
+
+def test_get_resource_types():
+    # Empty ament prefix will raise env error
+    set_ament_prefix_path([''])
+    try:
+        get_resource_types()
+    except EnvironmentError:
+        pass
+    except Exception as exc:
+        assert False, 'Expected EnvironmentError, got: {}'.format(type(exc))
+
+    set_ament_prefix_path(['prefix1', 'prefix2'])
+    resources = get_resource_types()
+    assert resources == {'resource_type1',
+                         'resource_type2',
+                         'resource_type3',
+                         'resource_type4',
+                         'resource_type5',
+                         'packages'}, 'Expected resources to ' + \
+        'be found: resource_type1,  resource_type2, ' + \
+        'resource_type3,  resource_type4, ' + \
+        'resource_type5 and packages'
+
+    set_ament_prefix_path(['prefix1'])
+    resources = get_resource_types()
+    assert resources == {'resource_type1',
+                         'resource_type2',
+                         'resource_type4',
+                         'resource_type5',
+                         'packages'}, 'Expected resources to ' + \
+        'be found: resource_type1,  resource_type2, ' + \
+        'resource_type4, resource_type5 and packages'
+
+
+def test_main_tool(capsys):
+    set_ament_prefix_path(['prefix1', 'prefix2'])
+    base_path = os.path.dirname(__file__)
```

Comment:
Prefer using [pathlib](https://docs.python.org/3/library/pathlib.html).
```suggestion
    base_path = Path(__file__)
```

With that, you can later use `base_path / 'sub_dir' / 'file'`.